graphql-wai/src/GraphQL/Wai.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module GraphQL.Wai
+  ( toApplication
+  ) where
+
+import Protolude
+
+import GraphQL (interpretAnonymousQuery)
+import GraphQL.Resolver (HasResolver, Handler)
+import Network.Wai (Application, queryString, responseLBS)
+import GraphQL.Value.ToValue (toValue)
+import Network.HTTP.Types.Header (hContentType)
+import Network.HTTP.Types.Status (status200, status400)
+import qualified Data.Aeson as Aeson
+
+
+-- | Adapt a GraphQL handler to a WAI application. This is really just
+-- to illustrate the mechanism, and not production ready at this point
+-- in time.
+--
+-- If you have a 'Cat' type and a corresponding 'catHandler' then you
+-- can use "toApplication @Cat catHandler".
+toApplication :: forall r. (HasResolver IO r) => Handler IO r -> Application
+toApplication handler = app
+  where
+    app req respond =
+      case queryString req of
+        [("query", Just query)] -> do
```

Comment:
I don't think the docs say you should handle both, the way I read them is "if it's a GET, here's how you do it, if it's a POST, here's how you do it"